Makefile

```diff
@@ -25,13 +25,43 @@ COMMUNITY       ?= true
 
 export MAIN_DIR OUT_DIR MAISTRA_BRANCH MAISTRA_VERSION COMMUNITY
 
+PACKAGE_NAME := github.com/maistra/istio-operator
+LD_EXTRAFLAGS =
+
+VERSION ?= development
+LD_EXTRAFLAGS += -X ${PACKAGE_NAME}/pkg/version.buildVersion=${VERSION}
+
+GITREVISION ?= $(shell git rev-parse --verify HEAD 2> /dev/null)
+ifeq ($(GITREVISION),)
+  GITREVISION = unknown
+endif
+LD_EXTRAFLAGS += -X ${PACKAGE_NAME}/pkg/version.buildGitRevision=${GITREVISION}
+
+GITSTATUS ?= $(shell git diff-index --quiet HEAD --  2> /dev/null; echo $$?)
```

Comment:
What's the purpose of this field? This is information that end-users are going to see, I think it would be best to omit things that can cause confusion.

This could be a check in the build script that would refuse to proceed in the presence of uncommitted changes Doesn't rhpkg/fedpkg do that already?.

Reply:
This is mostly informational. No checks or blockers. It's mimic'ing istio upstream output for their `version` commands.

Reply:
Hmm, ok. I'm not convinced this is a useful field, but if you think it's important to have it, I'm ok with it.

Reply:
I don't have a strong opinion. It's nice to have, but if you think it will cause more harm than good I can remove it quickly.

Reply:
I don't think it's a big deal. We can remove it if we find it causes confusion.

Reply:
I'll vote a yes to keeping it. Shows you what commit your on and is consistent with other software.